internal/api/impl/auth/auth.go

```diff
@@ -28,6 +30,19 @@ import (
 	"github.com/perses/perses/pkg/model/api/config"
 )
 
+func getRedirectURI(r *http.Request, authKind string, slugID string) string {
+	rd := url.URL{}
+	rd.Host = util.GetRequestHost(r)
+	rd.Scheme = util.GetRequestProto(r)
```

Comment:
I think importing this package just to extract the request host and the request proto is not worth it. The complexity of these methods is really low to avoid a copy and past